### ISSUE TITLE
Error "The LINQ expression node type 'Invoke' is not supported" (broken in >= 3.3)

### ISSUE THREAD

**Issue by ramondeklein** [MapsterMapper/Mapster#178]
We have a class `DbSecUser` that is mapped to `UserDto` and when using the `IQueryable<TDestination> ProjectToType<TDestination>(this IQueryable source, TypeAdapterConfig config = null)` method from `Mapster.Extensions` we get an exception with the message "**The LINQ expression node type 'Invoke' is not supported**".

We didn't touch our code, but we updated Mapster from v3.2 to v3.3.2 so we suspected Mapster has an issue. Downgrading Mapster to v3.2 immediately fixed the error.

This is our source type (I have removed al Entity Framework annotations):
```
    public class DbSecUser : IModelBase
    {
        public Guid Id { get; set; }
        public virtual DbSecCompany Company { get; set; }
        public Guid Company_Ref { get; set; }
        public string Email { get; set; }
        public string Description { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PasswordHash { get; set; }
        public bool IsActive { get; set; }
        public DateTime? ExpirationDate { get; set; }
        public string PhoneNumber { get; set; }
        public long? TelegramId { get; set; }
        public DateTime? LastInvitation { get; set; }
        public DateTime? UserCreated { get; set; }
        public string UserRole { get; set; }
        public DateTime? PasswordLastChanged { get; set; }
        public Guid? PasswordChangedByTokenId { get; set; }
        public string PasswordResetKey { get; set; }
        public DateTime? PasswordResetExpirationDate { get; set; }
        public string PreferredLocale { get; set; }

        public virtual ICollection<DbActGenerating> Generatings { get; set; } = new HashSet<DbActGenerating>();
        public virtual ICollection<DbRimClientConnection> Connections { get; set; } = new HashSet<DbRimClientConnection>();
        public virtual ICollection<DbRimNotificationGroupUser> NotificationGroupUsers { get; set; } = new HashSet<DbRimNotificationGroupUser>();
        public virtual ICollection<DbSecUserGroupAssignment> GroupAssignments { get; set; } = new HashSet<DbSecUserGroupAssignment>();
        public virtual ICollection<DbSecUserSystemGroupAssignment> SystemGroupAssignments { get; set; } = new HashSet<DbSecUserSystemGroupAssignment>();
        public virtual ICollection<DbSvcBridgeLogging> BridgeLogging { get; set; } = new HashSet<DbSvcBridgeLogging>();
        public virtual ICollection<DbWebUserSession> Sessions { get; set; } = new HashSet<DbWebUserSession>();

        public string GetPreferredLocale() => PreferredLocale ?? Company.PreferredLocale;
    }
```

And these are our destination types:
```
    public class UserBriefDto
    {
        public Guid Id { get; set; }
        public Guid CompanyId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName => $"{FirstName} {LastName}";
        public string EmailAddress { get; set; }
        public long? TelegramId { get; set; }
    }

    public class UserDto : UserBriefDto
    {
        public string PhoneNumber { get; set; }
        public string Role { get; set; }
        public bool IsActive { get; set; }
        public DateTime? LastInvitation { get; set; }
        public DateTime? ExpirationDate { get; set; }
        public UserSessionDto Session { get; set; }
        public IEnumerable<GroupDto> Groups { get; set; } = Enumerable.Empty<GroupDto>();
        public bool AllowImpersonation { get; set; }
        public string PreferredLocale { get; set; }
    }
```

The mapping is defined like this:
```
        config.ForType<DbSecUser, UserBriefDto>()
            .Map(i => i.FullName, i => $"{i.FirstName} {i.LastName}")
            .Map(i => i.EmailAddress, i => i.Email)
            .Map(i => i.CompanyId, i => i.Company_Ref);

        config.ForType<DbSecUser, UserDto>()
            .Inherits<DbSecUser, UserBriefDto>()
            .Map(i => i.Role, i => i.UserRole)
            .Map(i => i.LastInvitation, i => i.LastInvitation)
            .Map(i => i.Session, i => i.Sessions.FirstOrDefault(j => j.EndUtc == null))
            .Map(i => i.Groups, i => i.GroupAssignments.Where(j => j.Group != null).Select(j => new { Id = j.Group_Ref, Name = j.Group.Description }));
```

**Comment by chaowlert** [comment]
Thank you for the detailed config. I will fix soon.

**Comment by chaowlert** [comment]
Sorry for late, I fixed it as part of 4.0.
